dart/constraint/BoxedLcpConstraintSolver.hpp

```diff
@@ -116,6 +116,15 @@ class BoxedLcpConstraintSolver : public ConstraintSolver
   std::vector<s_t*> solveConstrainedGroup(
       ConstrainedGroup& group, simulation::World* world) override;
 
+  /// Build the inputs to the LCP from the constraint group.
+  bool buildLcpInputs(ConstrainedGroup& group);
```

Comment:
I think we should consider renaming this function. When I first read the name, I expected it to return a struct with all the LCP inputs (`A` matrix, `b` vector, `upperBound` vector, `lowerBound` vector, `fIndex` vector). But it returns a boolean. Maybe `shouldBuildLcpInputs()`? Or something...

Reply:
Also, actually, this suggests the idea of a method that returns the LCP inputs, which is what I thought this was at first. `struct LcpInputs buildLcpInputs(ConstraintGroup& group)`, where we have

```
struct LcpInputs {
Eigen::MatrixXs A;
Eigen::VectorXs b;
Eigen::VectorXs upperBound;
Eigen::VectorXs lowerBound;
Eigen::VectorXi fIndex;
};
```

What do you think? The downside of this change is it's just more surface area for people to have to learn about in the APIs to get things to work. The alternative is to leave it as is, and hide the setup of the LCP inside of `solveLcp`, which has certain advantages for simplicity.